### ISSUE TITLE
Migrate from `TransactionSigned` to `alloy_consensus::Signed<T>`

### ISSUE THREAD

**Issue by emhane** [paradigmxyz/reth#11510]
### Describe the feature

Remove reth type 

https://github.com/paradigmxyz/reth/blob/36de90fdc346bc3aacbfb311fa20b623d48ef867/crates/primitives/src/transaction/mod.rs#L1035-L1047

in favour of `alloy_consensus::Signed<T, S>`. In order to limit scope, it's recommended to introduce type alias

```rust
pub type TransactionSigned = Signed<reth_primitives::Transaction, alloy_primitives::Signature>;
```

blocked by https://github.com/paradigmxyz/reth/issues/11253

**Comment by emhane** [comment]
Starting with turning `TransactionSigned` into a wrapper of alloy type should unblock this
```rust
#[derive(derive_more::Deref)]
pub struct TransactionSigned {
    #[deref]
    inner: Signed<Transaction, Signature>
}
```

**Comment by klkvr** [comment]
this is a bit different from how alloy handles transaction enums: https://github.com/alloy-rs/alloy/blob/de2a13340303db6dde1296f68df57078a5b1fdd8/crates/consensus/src/transaction/envelope.rs#L93-L115

there might be valid transactions with no signatures, eg right now we keep a dummy signature for deposit tx which is never encoded

should we consider refactoring `TransactionSigned` into an enum or embedding alloy's envelope right away?

**Comment by emhane** [comment]
pls link if a conflicting issue exists @klkvr 

I don't see how fixing this issue will block using `TxEnvelope` later. It will implement the methods of `TransactionSigned` in the way possible on `Signed<Transaction>`, which will be used still within `TxEnvelope` which wraps `Signed<T>`.

**Comment by klkvr** [comment]
> pls link if a conflicting issue exists @klkvr

I don't think so. Though this conflicts a bit with part of alloy's `Signed<T>` API which requires inner tx to be signable https://github.com/alloy-rs/alloy/blob/48be08ff0bea3ce49f09ac2d60aaaa3c850dda0b/crates/consensus/src/signed.rs#L46, which is not satisfied by our transaction enum

> It will implement the methods of `TransactionSigned` in the way possible on `Signed<Transaction>`, which will be used still within `TxEnvelope` which wraps `Signed<T>`.

`TxEnvelope` wraps individual tx variants and not the entire typed transaction enum, so this doesn't get `TransactionSigned` structure closer to envelopes. Can't we already target for replacing `TransactionSigned` with respective envelopes from alloy and op-alloy? looks like we are pretty close to it and only some recovery helpers would need to be moved

**Comment by emhane** [comment]
> > pls link if a conflicting issue exists @klkvr
> 
> I don't think there are any. However this conflicts a bit with part of alloy's `Signed<T>` API which requires inner tx to be signable https://github.com/alloy-rs/alloy/blob/48be08ff0bea3ce49f09ac2d60aaaa3c850dda0b/crates/consensus/src/signed.rs#L46, which is not satisfied by our transaction enum
> 
> > It will implement the methods of `TransactionSigned` in the way possible on `Signed<Transaction>`, which will be used still within `TxEnvelope` which wraps `Signed<T>`.
> 
> `TxEnvelope` wraps individual tx variants and not the entire typed transaction enum, so this doesn't get `TransactionSigned` structure closer to envelopes. Can't we already target for replacing `TransactionSigned` with respective envelopes from alloy and op-alloy? looks like we are pretty close to it and only some recovery helpers would need to be moved

is the signature type different for op?

**Comment by klkvr** [comment]
> is the signature type different for op?

not every op transaction has a signature, but for variants that have we use the same signature type

**Comment by emhane** [comment]
for reth integration, op will need a new signature type. in order to implement reth traits on that signature, the type can be nested inside of the `TxEnvelope` variants, but it makes more sense for the signature to be parallel with the tx type because the signature type doesn't have anything directly to do with the tx type = envelope type.

tldr; we can't use the `TxEnvelope` type as is. we need it to wrap the unsigned tx.

**Comment by emhane** [comment]
the main reason we want to do this, is because then we can rely on alloy trait `SignableTransaction`, extending the common surface between reth and other ethereum libraries depending on alloy
https://github.com/alloy-rs/alloy/blob/ef629fc61fff2453847202c14af7a5677f979f88/crates/consensus/src/transaction/mod.rs#L193-L197

**Comment by emhane** [comment]
> imo this doesn't make a lot of sense because SignableTransaction is client POV and we don't need this for reth, since we don't do any signing.

ok, makes sense

> this also has the same problem as `Sealed` because we can't implement foreign traits on this

it would have to be the inner type of a wrapper type

still, the methods of `Signed<T>` are usable in reth, and extending common surface between reth and alloy is beneficial for interop of rust ethereum ecosystem, since many libraries are using alloy that maybe aren't using reth yet. the more common code, the easier it will be to plug into reth at some later point.

**Comment by emhane** [comment]
actually @mattsse , on second look, we use the `Signable` trait methods not only for signing, also for sender recovery

**Comment by mattsse** [comment]
I'm marking this as complete, we upstreamed more tx types to alloy (Pooled) and are now reusing Signed where we can,

we can't fully use the `Signed` because the hash isn't lazy
